lib/rules/no-unregistered-components.js

```diff
@@ -178,7 +184,20 @@ module.exports = {
               })
             )
         }
-      },
+      }),
+      utils.defineVueVisitor(context, {
+        /** @param {ObjectExpression} obj */
+        onVueObjectEnter(obj) {
+          const nameProperty = utils.findProperty(obj, 'name')
+
+          if (nameProperty && ignoreRecursive) {
+            registeredComponents.push({
+              node: nameProperty,
+              name: nameProperty.value.value
+            })
+          }
+        }
+      }),
       utils.executeOnVue(context, (obj) => {
         registeredComponents.push(...utils.getRegisteredComponents(obj))
       })
```

Comment:
Could you please change it as follow?
Also, remove `utils.compositingVisitors`.

```js
      utils.executeOnVue(context, (obj) => {
        registeredComponents.push(...utils.getRegisteredComponents(obj))
        
          const nameProperty = utils.findProperty(obj, 'name')

          if (nameProperty) {
            registeredComponents.push({
              node: nameProperty,
              name: nameProperty.value.value
            })
          }
      })
```